src/filters/FilterSystem.ts

```diff
@@ -68,17 +68,83 @@ export interface FilterInstruction extends Instruction
     filterEffect: FilterEffect,
 }
 
-export interface FilterData
+/**
+ * Class representing the data required for applying filters.
+ * This class holds various properties that are used during the filter application process.
+ */
+class FilterData
 {
-    skip: boolean;
-    enabledLength?: number;
-    inputTexture: Texture
-    bounds: Bounds,
-    blendRequired: boolean,
-    container: Container,
-    filterEffect: FilterEffect,
-    previousRenderSurface: RenderSurface,
-    backTexture?: Texture,
+    /**
+     * Indicates whether the filter should be skipped.
+     * @type {boolean}
+     */
+    public skip = false;
+
+    /**
+     * The texture to which the filter is applied.
+     * @type {Texture}
+     */
+    public inputTexture: Texture = null;
+
+    /**
+     * The back texture used for blending, if required.
+     * @type {Texture | undefined}
+     */
+    public backTexture?: Texture = null;
+
+    /**
+     * The list of filters to be applied.
+     * @type {Filter[]}
+     */
+    public filters: Filter[] = null;
+
+    /**
+     * The bounds of the filter area.
+     * @type {Bounds}
+     */
+    public bounds = new Bounds();
+
+    /**
+     * The container to which the filter is applied.
+     * @type {Container}
+     */
+    public container: Container = null;
+
+    /**
+     * Indicates whether blending is required for the filter.
+     * @type {boolean}
+     */
+    public blendRequired: boolean = false;
+
+    /**
+     * The render surface where the output of the filter is rendered.
+     * @type {RenderSurface}
+     */
+    public outputRenderSurface: RenderSurface = null;
+
+    /**
+     * The offset of the output render surface.
+     * @type {{ x: number, y: number }}
```

Comment:
We should use the point data type for x/y values to keep it consistent with the rest of pixi

```suggestion
     * @type {PointData}
```